### PR TITLE
docs: create guide for primitive data type

### PR DESCRIPTION
**Pull request by bwanglzu**
Created a screenshot to view:

need some insights from @FionnD where should we put the guide.

FireShot Capture 003 - A guide on Jina Primitive Data Types — Jina 0 9 28 documentation - 127 0 0 1

### CODE REVIEW

File: chapters/primitive_data_type.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
+=======================================
+A guide on Jina Primitive Data Types
+=======================================
+
+.. meta::
+   :description: A guide on Jina Primitive Data Types
+   :keywords: Jina, primitive data types
+
+.. note:: This guide expect you have a basic understanding of Jina, if you havn't, please checkout `Jina 101 <https://docs.jina.ai/chapters/101/index.html>`_ first.
```
Comment (cristianmtr) [inline chapters/primitive_data_type.rst]:
```suggestion
.. note:: This guide expect you have a basic understanding of Jina, if you haven't, please check out `Jina 101 <https://docs.jina.ai/chapters/101/index.html>`_ first.
```

Comment (bwanglzu) [reply chapters/primitive_data_type.rst]: thanks! @cristianmtr

File: chapters/primitive_data_type.rst [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+Before you start
+====================
+
+We expect you have a clean Python 3.7/3.8/3.9 (virtual) build.
+WIth Jina installed on your machine:
+
+.. highlight:: bash
+.. code-block:: bash
+
+    pip install -U jina>=0.8
```
Comment (JoanFM) [inline chapters/primitive_data_type.rst]: I would just assume people install something newer and assume they did this step?

**Comment by bwanglzu** [comment]
@FionnD @davidbp ~can you take a look from the "content" perspective? I'm still working on the grammar things~ ready to review

**Review by JoanFM — COMMENTED** [review]
Really good document! just minor comment

File: chapters/primitive_data_type.rst [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+   * - QueryLang
+     - basic
+     - A Pythonic interface to access and manipulate :class:`QueryLangProto`.
+   * - QueryLangSet
+     - composite
+     - A mutable sequence of :class:`QueryLang`.
+   * - Request
+     - basic
+     - A Pythonic interface to access and manipulate :class:`RequestProto`.
+   * - NameScore
```
Comment (JoanFM) [inline chapters/primitive_data_type.rst]: is It NameScore or `NamedScore`?

**Review by CatStark — APPROVED** [review]

**Comment by CatStark** [comment]
Great work @bwanglzu !

**Comment by FionnD** [comment]
> @FionnD @davidbp can you take a look from the "content" perspective? I'm still working on the grammar things

From a content perspective lgtm, if you want a grammar review hit me up

File: chapters/primitive_data_type.rst [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+Before we introduce the Jina primitive data types, ``drivers`` helps the ``executors`` to handle the network traffic by directly interacting with the Protobuf messages.
+Thus our back-end engineers have to generate or parse stream of bytes in the network layer.
+This is not aligned with the design principle of Jina.
+
+
+Before you start
+====================
+
+We expect you have a clean Python 3.7/3.8/3.9 (virtual) build.
+WIth Jina installed on your machine:
```
Comment (CatStark) [inline chapters/primitive_data_type.rst]: Small typo "with"

File: chapters/primitive_data_type.rst (hunk 2 above)
Comment (CatStark) [reply chapters/primitive_data_type.rst]: I'm very skeptical about assuming anything in tech

Comment (bwanglzu) [reply chapters/primitive_data_type.rst]: okay, I'll remove `>=0.8`, let's keep this in section `Before you start`

File: chapters/primitive_data_type.rst (hunk 4 above)
Comment (bwanglzu) [reply chapters/primitive_data_type.rst]: thanks, fixed!

File: chapters/primitive_data_type.rst (hunk 3 above)
Comment (bwanglzu) [reply chapters/primitive_data_type.rst]: good catch, fixed!

File: chapters/primitive_data_type/primitive_data_type.rst [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+    # Check the modalities of the document
+    print(multimodal_document.modalities)
+    # Get the content of document by modality name
+    content = multimodal_document['visual']
+
+
+Design Decisions
+====================
+
+While designing and implementing Jina primitive data types, we have been always keep the following principles in mind:
```
Comment (FionnD) [inline chapters/primitive_data_type/primitive_data_type.rst]: kept not keep

File: chapters/primitive_data_type/primitive_data_type.rst [hunk 6]
```diff
@@ -0,0 +1,282 @@
+=======================================
+A guide on Jina Primitive Data Types
+=======================================
+
+.. meta::
+   :description: A guide on Jina Primitive Data Types
+   :keywords: Jina, primitive data types
+
+.. note:: This guide expect you have a basic understanding of Jina, if you haven't, please check out `Jina 101 <https://docs.jina.ai/chapters/101/index.html>`_ first.
```
Comment (FionnD) [inline chapters/primitive_data_type/primitive_data_type.rst]: I would replace expect with assumes.

**Review by FionnD — CHANGES_REQUESTED** [review]
This is a **really** well-written guide Bo! A great example for the team for the standard we are looking for :)

**Review by FionnD — APPROVED** [review]

**Pull request merged**